### ISSUE TITLE
Nest CLI Assets Property Fails to Copy Files on Windows Due to Path Format

### ISSUE THREAD

**Issue by eranbetzalel** [nestjs/nest-cli#2397]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using the assets property in nest-cli.json, the file copying returns 0 files on Windows. This seems to be caused by `path.join ` converting paths to Windows-style backslashes, which the glob package does not recognize.

I've debugged the code and saw this specific behavior in the copyAssets function.

```
{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "assets": ["**/assets/*.html"]
  }
}
```

### Minimum reproduction code

Not available, issue observable in the behavior of nest-cli

### Steps to reproduce

Define assets in nest-cli.json using glob patterns.
Run a build or start command that triggers assets copying.
Observe that no files are copied.

### Expected behavior

The assets defined in nest-cli.json should be correctly copied regardless of the OS path style.

### Package version

10.2.1

### NestJS version

10.2.8

### Node.js version

18.17.1

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
please share the output of `npx nest info`

**Comment by eranbetzalel** [comment]
```
[System Information]
OS Version     : Windows 10
NodeJS Version : v18.15.0
NPM Version    : 9.5.0     
                           
[Nest CLI]                 
Nest CLI Version : 10.2.1  
                           
[Nest Platform Information]
platform-express version : 10.2.10
cache-manager version    : 2.1.1  
event-emitter version    : 2.0.3
microservices version    : 10.2.10
mapped-types version     : 2.0.4
schematics version       : 10.0.3
passport version         : 10.0.2
terminus version         : 10.1.1
mongoose version         : 10.0.1
common version           : 10.2.10
config version           : 3.1.1
core version             : 10.2.10
jwt version              : 10.2.0
cli version              : 10.2.1
 version                 : 4.0.1
 version                 : 3.1.1
 version                 : 3.0.1

```

**Comment by eranbetzalel** [comment]
This is a valid workaround for the development environment.

Adding `"watchAssets": true` seems to make it work correctly.

```
{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "assets": [
      {
        "include": "**/assets/*.html",
        "watchAssets": true
      }
    ]
  }
}
```

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue? 🙏

**Comment by eranbetzalel** [comment]
@kamilmysliwiec, Sure, I can do that.

**Comment by kasium** [comment]
@eranbetzalel had you time to create a  PR? This is quite bad issue for me

**Comment by CustomEntity** [comment]
@kamilmysliwiec Is this a good "first issue"? I'd like to create a PR to fix this

**Comment by kamilmysliwiec** [comment]
Feel free to do so @CustomEntity

**Comment by kamilmysliwiec** [comment]
Fixed in 10.4.0

**Comment by eranbetzalel** [comment]
Thank you @CustomEntity.
